### PR TITLE
fix: thread `enforce_ulimit_nofile` config down when opening blockstore

### LINKED ISSUE #23615: web3.js: CI tests failing
#### Problem

Web3.js CI tests are failing in master, which is seen on unrelated PRs e.g. https://github.com/solana-labs/solana/pull/23579

@jacobcreech on [Discord](https://discord.com/channels/428295358100013066/476811830145318912/951886952372772929):
> There's a bunch of npm test failures on test:live with similar messages to Error: failed to get confirmed block: Block 1 cleaned up, does not exist on node. First available block: 2
> They're failing with these on the local validator from the looks of it

#### Proposed Solution

> There needs to be some error handling from the looks of it on this while loop in the tests https://github.com/solana-labs/solana/blob/master/web3.js/test/connection.test.ts#L1700-L1708
> That or longer storage enabled on the local validator

### PR DESCRIPTION
**Pull request by steveluscher**
#### Problem

The web3.js test suite starts up a test validator in a CI environment (GitHub Actions workflows). That environment has a lower maximum allowable file descriptors than the validator would like.

https://github.com/solana-labs/solana/blob/c83c95b56b375afeea231b45cb4ced0cf07761e8/ledger/src/blockstore.rs#L4273-L4276

There's a config param that decides whether a shortfall in the file descriptor limit should log and panic, or just log and move on.

https://github.com/solana-labs/solana/blob/c83c95b56b375afeea231b45cb4ced0cf07761e8/ledger/src/blockstore.rs#L4304-L4306

That config param was not being threaded through when opening the blockstore, so no matter the value, the test validator would _always_ panic in the CI environment and the test suite would fail.

This blocks the release of new versions of web3.js.

#### Summary of Changes

Add  `enforce_ulimit_nofile` to the `BlockstoreOptions` when starting a validator.

Fixes #23615.

---
NOTE: I don't expect the CI run on this PR to actually pass. The CI run sources the `edge` test validator, rather than the version of the test validator _in this PR itself_. The web3.js tests will start passing once this PR is merged and hits `edge`.

### CODE REVIEW

**Review by steviez — APPROVED** [review]
Looks good to me! And to potentially same anyone else who wanted to look these up, the full validator leaves this flag unspecifed
https://github.com/solana-labs/solana/blob/9cf772092273c98fa35cd9a2f23d635f47eac6aa/validator/src/main.rs#L2450
so it will fall back on default
https://github.com/solana-labs/solana/blob/9cf772092273c98fa35cd9a2f23d635f47eac6aa/core/src/validator.rs#L193

which matches default in `BlockstoreOptions`:
https://github.com/solana-labs/solana/blob/9cf772092273c98fa35cd9a2f23d635f47eac6aa/ledger/src/blockstore_db.rs#L1564

whereas the test validator is explicitly setting the flag
https://github.com/solana-labs/solana/blob/9cf772092273c98fa35cd9a2f23d635f47eac6aa/test-validator/src/lib.rs#L691

So, full validator behavior remains unchanged. As a separate aside, I think we could make `ValidatorConfig` contain a `BlockstoreOptions` instead of the individual options to avoid duplicated default values. I can do that quick cleanup after this PR goes in to avoid merge conflict

**Comment by steveluscher** [comment]
Sweet! Can you ship this @steviez? I accidentally stripped myself of repro privileges in GitHub, and now I have to wait for someone to approve my team join request :)

**Comment by steviez** [comment]
@steveluscher Done and was wondering why the community tag got added in.

**Comment by lijunwangs** [comment]
This was recent regression when BlockstoreOptions was introduced. It blocked me from running a local cluster with low open file count limit. https://github.com/solana-labs/solana-accountsdb-plugin-postgres/runs/5701299256?check_suite_focus=true

Marked it to be merged to v1.10

**Pull request merged**
